specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1409 @@
+is called or process termination.
+
+### 5.4.2. Boostrapper - Find and Load ProjectReunion.dll
+
+The bootstrapper API finds the ProjectReunion Framework package and calls LoadLibrary("ProjectReunion.dll")
+to load and initialize the Dynamic Dependencies API.
+
+## 5.5. Dynamic Dependencies vis a vis Static Dependencies
+
+Dynamic dependencies' package dependency is equivalent to an MSIX package's `<PackageDependency>` in
+`appxmanifest.xml`. This defines criteria to be resolved to a package for use at runtime in a
+process' package graph.
+
+Packages dynamically added to a package graph can be removed. Packages in the package graph at
+process creation (aka static package graph) cannot be removed; the static package graph is a part of
+a process' DNA, from birth to death.
+
+## 5.6. Known Limitations in v1
+
+The initial implementation does not fully implement this spec.
+
+### 5.6.1. [uap6:LoaderSearchPathOverride](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-loadersearchpathoverride) not supported
+
+See [3.2.1. Known Issue: DLL Search Order ignores uap6:LoaderSearchPathOverride](#321-known-issue-dll-search-order-ignores-uap6loadersearchpathoverride).
+
+### 5.6.2. [uap6:AllowExecution](https://docs.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-uap6-allowexecution) not supported
+
+See [3.2.2. Known Issue: DLL Search Order ignores uap6:AllowExecution](#322-known-issue-dll-search-order-ignores-uap6allowexecution).
+
+# 6. API Details
+
+**TODO** Retitle this section 'API' since we don't have 'API Notes'?
+
+## 6.1. Win32 API - MsixDynamicDependency.hpp
+
+All Win32 APIs are prefixed with Mdd/MDD for MSIX Dynamic Dependencies.
+
+```c++
+enum class MddPinPackageDependency : uint32_t
+{
+    None 0,
+
+    /// Disable dependency resolution when pinning a package dependency.
+    DoNotVerifyDependencyResolution = 0x00000001,
+
+     /// Define the package dependency for the system, accessible to all users
+     /// (default is the package dependency is defined for a specific user).
+     /// This option requires the caller has adminitrative privileges.
+    ScopeIsSystem = 0x00000002,
+};
+DEFINE_ENUM_FLAG_OPERATORS(MddPinPackageDependency)
+
+enum class MddPinPackageDependencyLifetimeKind
+{
+    /// The current process is the lifetime artifact. The package dependency
+    /// is implicitly unpinned when the process terminates.
+    Process = 0,
+
+    /// The lifetime artifact is an absolute filename or path.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    FilePath = 1,
+
+    /// The lifetime artifact is a registry key in the format
+    /// 'root\\subkey' where root is one of the following: HKLM, HKCU, HKCR, HKU.
+    /// The package dependency is implicitly unpinned when this is deleted.
+    RegistryKey = 2,
+}
+
+enum class MddAddPackageDependency : uint32_t
```

Comment:
Compiler won't accept same named enum+function e.g. `enum foo{...}` and `void foo();` or at least not `extern "C" void foo();`

Rename to `MddAddPackageDependencyOptions` or `MddAddPackageDependencyFlags`